File: sdk/js/packages/core/src/domain/event.ts
/**
 * Pure data describing one telemetry event at capture time.
 * No methods, no I/O — only what the application can know.
 * The gateway adds org context, enrichment, and ingestion timestamp.
 */
export interface EventInput {
  readonly eventType: string;
  readonly severity: string;
  readonly message: string;
  readonly userId?: string;
  readonly sessionId?: string;
  readonly traceId?: string;
  readonly spanId?: string;
  readonly parentSpanId?: string;
  readonly payload?: Record<string, unknown>;
}

/** Options shared across all typed capture methods. */
export interface CaptureOptions {
  userId?: string;
  sessionId?: string;
  traceId?: string;
  spanId?: string;
  parentSpanId?: string;
  payload?: Record<string, unknown>;
}

File: sdk/python/src/watcher_sdk/domain/event.py
# Pure data that the application passes to the SDK.
# No I/O, no dependencies — only what we know at capture time.
# The gateway adds org context, enrichment metadata, and ingestion timestamp.
from dataclasses import dataclass, field
from typing import Any


@dataclass(frozen=True)
class EventInput:
    """Everything the SDK needs to build one telemetry event."""

    event_type: str   # audit | log | trace | metric | event
    severity: str     # debug | info | warn | error | critical
    message: str
    user_id: str = ""
    session_id: str = ""
    trace_id: str = ""
    span_id: str = ""
    parent_span_id: str = ""
    payload: dict[str, Any] = field(default_factory=dict)
